Updated IntegrationTestHelper to mock all api functions

rhysyngsun opened the pull request:
#### What are the relevant tickets?
N/A

#### What's this PR do?
Refactors `IntegrationTestHelper` to stub out all the API methods.

#### How should this be manually tested?
Test should pass

noisecapella commented on static/js/util/integration_test_helper.js:
```diff
@@ -36,12 +36,16 @@ export default class IntegrationTestHelper {
-    this.createCommentStub.throws(new Error("not implemented"))
+    // stub all the api functions
+    // NOTE: due to a regression in sinon 2.x, if you use `callsFake` you must
+    //       call `resetBehavior()` on the stub first or the error still raises
```
I checked and it looks like this is fixed in sinon 3.x FYI

rhysyngsun replied:
I tried upgrading and these still fail when I remove the `resetBehavior` calls

noisecapella commented:
Looks like there are some errors in the PR tests

rhysyngsun commented:
PR tests were failing because this commit added more hardcoded stubs: https://github.com/mitodl/open-discussions/commit/5ad04499d791d31a35312e6d4ec57189665c9209#diff-fd6fb1b588e6057629caba91f07d6dbbR46

I just rebased and updated the one commit to remove those as well. I'll merge when tests pass again.
